### PR TITLE
Coincatch integration

### PR DESCRIPTION
**Pull request by rayBastard**
#22928

### CODE REVIEW

**Review by Rokdima — COMMENTED** [review]
ts/src/abstract/coincatch.ts

File: ts/src/coincatch.ts [hunk 1]
```diff
@@ -0,0 +1,5299 @@
...
+            if (timeInForce !== undefined) {
+                request['timeInForceValue'] = timeInForce;
+            }
+            let clientOrderId: Str = undefined;
+            [ clientOrderId, params ] = this.handleParamString (params, 'clientOrderId');
+            if (clientOrderId !== undefined) {
+                request['clientOid'] = clientOrderId;
+            }
+            if (type === 'limit') {
+                request['executePrice'] = price; // spot markets have no precision
```
Comment (carlosmiei) [inline ts/src/coincatch.ts:2268]: don't we need to convert price to string as we do with amount? (if it works as is then it is ok)

Comment (yzh-pelle) [reply ts/src/coincatch.ts:2268]: Yes, it works good

File: ts/src/coincatch.ts [hunk 2]
```diff
@@ -0,0 +1,5299 @@
...
+        if (isMarketBuy) {
+            const costAndParams = this.handleRequiresPriceAndCost (methodName, params, price, amount, cost);
+            cost = costAndParams['cost'];
+            params = costAndParams['params'];
+        }
+        if (triggerPrice === undefined) {
+            if (type === 'limit') {
+                request['price'] = price; // spot markets have no precision
+            }
+            request['quantity'] = isMarketBuy ? cost : amount.toString (); // spot markets have no precision
```
[the remark this answers is not in the document]
Comment (rayBastard) [reply ts/src/coincatch.ts]: [db85606](https://github.com/ccxt/ccxt/pull/23589/commits/db856066e39069a5a2f0aafe72629db94d3b6455)

fixed

**Comment by carlosmiei** [comment]
coincatchLogo

**Pull request merged**